internal/controllers/gateway/gateway_utils_test.go

```diff
@@ -130,15 +130,14 @@ func TestGetListenerStatus_no_duplicated_Detached_condition(t *testing.T) {
 	require.NoError(t, err)
 	require.Len(t, statuses, 1, "only one listener status expected as only one listener was defined")
 	listenerStatus := statuses[0]
-	assertOnlyOneConditionOfType(t, listenerStatus.Conditions, gatewayv1beta1.ListenerConditionAccepted)
+	assertOnlyOneConditionForType(t, listenerStatus.Conditions)
 }
 
-func assertOnlyOneConditionOfType(t *testing.T, conditions []metav1.Condition, typ gatewayv1beta1.ListenerConditionType) {
-	conditionNum := 0
-	for _, condition := range conditions {
-		if condition.Type == string(typ) {
-			conditionNum++
-		}
+func assertOnlyOneConditionForType(t *testing.T, conditions []metav1.Condition) {
+	conditionsNum := lo.CountValuesBy(conditions, func(c metav1.Condition) string {
+		return c.Type
+	})
+	for _, n := range conditionsNum {
+		assert.Equal(t, 1, n)
```

Comment:
Nit: I'd still suggest to add a descriptive error message in case this assertion fails. WDYT?

Reply:
sure, I missed that :)